l10n_be_cooperator_national_number/models/res_partner.py

```diff
@@ -2,25 +2,41 @@
 #
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
-from odoo import models
+from odoo import api, models
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    @api.model
+    def _get_belgian_id_category(self):
+        return self.env.ref(
+            "l10n_be_partner_identification.l10n_be_national_registry_number_category"
+        )
+
+    def get_belgian_national_number_id_number(self):
+        self.ensure_one()
+        belgian_cat = self._get_belgian_id_category()
+        return self.env["res.partner.id_number"].search(
+            [
+                ("partner_id", "=", self.id),
+                ("category_id", "=", belgian_cat.id),
+            ],
+            limit=1,
+        )
+
+    def get_belgian_national_number(self):
+        self.ensure_one()
+        id_number = self.get_belgian_national_number_id_number()
```

Comment:
these also exist on `subscription.request`. please move the code from there to here and use it.

Reply:
moved to here and `res.partner.id_category`